### ISSUE TITLE
Imported symbols are erroneously detected undefined when using JSX

### ISSUE THREAD

**Issue by n1k0** [babel/babel-eslint#17]
```js
import App from "./components/App";

React.render(<App />, document.querySelector("#app"));
```

Gives:

```
26:14  error  'App' is not defined  no-undef
```

Also,

```js
import App from "./components/App";

React.render(App(), document.querySelector("#app"));
```

makes it going away, so it's a JSX-related thingy.

**Comment by gaearon** [comment]
+1

**Comment by nzakas** [comment]
This is because ESLint doesn't yet support modules in our rules. See https://github.com/eslint/eslint/issues/1903

**Comment by kentcdodds** [comment]
@nzakas, has that changed in the last 13 days? Currently on the website there's the `ecmaFeatures.modules` option, so that's why I ask. But I'm still experiencing this issue. Current workaround was to use `const foo = require('./foo');` because I'm using webpack and it's cool like that, but would love to just use pure ES6 modues.

**Comment by nzakas** [comment]
Yup. We now support modules. I'd recommend either subscribing to the mailing list or following geteslint on Twitter for updates.

**Comment by kentcdodds** [comment]
I'm following on Twitter. Do you know why I would bet getting this error:

```javascript
import foo from 'foo';

foo.bar();
```

Results in:

```
error  'foo' is not defined  no-undef
```

I npm installed a commit from yesterday.

**Comment by nzakas** [comment]
If you think it's an error in ESLint (without babel-eslint), then please look at the issues over on the ESLint repo.

**Comment by sebmck** [comment]
Can you please verify that this issue still exists on 3.0.1?

**Comment by sebmck** [comment]
Not sure I can do anything about this. I'd recommend using the [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react) rules.
